apps/lib/apps.c

```diff
@@ -1894,7 +1906,7 @@ static const char *get_dp_url(DIST_POINT *dp)
         uri = GENERAL_NAME_get0_value(gen, &gtype);
         if (gtype == GEN_URI && ASN1_STRING_length(uri) > 6) {
             const char *uptr = (const char *)ASN1_STRING_get0_data(uri);
-            if (strncmp(uptr, "http://", 7) == 0)
+            if (strncmp(uptr, OSSL_HTTP_PREFIX, strlen(OSSL_HTTP_PREFIX)) == 0)
```

Comment:
Uhm, why not https?  You could simply use `is_http` here...

Reply:
We could do that, but it usually makes little sense to load CRLs via HTTPS. They are already self-contained w.r.t. protection.
One would even risk going into infinite recursion if CRLs are again be needed for securing the connection.

Reply:
Considering that there's been a huge push to have *everything* web secured with https (even static pages), and "letsencrypt makes it easy", we might actually get that surprise.

Reply:
Folks setting up CDPs hopefully know what they're doing.

Reply:
This sub-thread brings to mind a question: do all apps that might use this library take flags to pass the trust store for https, and do they pass it down to the library?

Reply:
To be explicit, I am asking how the app would verify a TLS cert without a known trust store?  If it doesn't check, then *that should be documented*

Reply:
Good point - also for any other use cases of `load_crl()` and `load_cert_pass()` the `https://` variant makes no sense because they do not offer the possibility to pass in a trust store nor any other TLS-specific data!

So I've changed the code to give a respective error message is this case.